aeson.cabal

```diff
@@ -162,6 +162,9 @@ test-suite tests
     vector,
     quickcheck-instances >=0.3.12
 
+  if !impl(ghc >= 8.0)
+    build-depends: semigroups >= 0.16.1
```

Comment:
@hvr is this a roundabout way of asking for an upper bound? If so, I'd happily add one and open a new PR so this can get in soon.